src/Text.mo

```diff
@@ -425,6 +425,54 @@ module {
     }
   };
 
+  /// Returns a substring of the input `Text` delimited by the specified `Pattern`, provided with a starting position and a length.
+  /// If no length is passed, returns empty string.
+  ///
+  /// ```motoko include=import
+  /// Text.substring("This is a sentence.", 0, 4) // "This"
+  /// Text.substring("This is a sentence.", 5, 4) // "is a"
+  /// Text.substring("This is a sentence.", 0, 0) // ""
+  /// ```
+  public func substring(t : Text, start : Int, len : Int) : Text {
```

Comment:
This was based off of the JavaScript patterns, where you are allowed to use negative indexes to count backward from the end of iterable structures. There's no need to introduce that to Motoko, however

Reply:
I think in my opinion it's nicer to return an empty string over `null`. I've defined those cases in the doc comment, where `start` or `length` exceeds the length of a base string, it will 
- return "" if start is beyond input.size()
- return the substring from the starting position to the end of the input if length exceeds the rest